x/leverage/genesis.go

```diff
@@ -16,11 +16,96 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 		k.SetRegisteredToken(ctx, token)
 	}
 
-	k.InitializeLastInterestTime(ctx)
+	for _, borrow := range genState.Borrows {
+		borrower, err := sdk.AccAddressFromBech32(borrow.Address)
+		if err != nil {
+			panic(err)
+		}
+
+		if err = k.SetBorrow(ctx, borrower, borrow.Amount); err != nil {
+			panic(err)
+		}
+	}
+
+	for _, setting := range genState.CollateralSettings {
+		borrower, err := sdk.AccAddressFromBech32(setting.Address)
+		if err != nil {
+			panic(err)
+		}
+
+		if err = k.SetCollateralSetting(ctx, borrower, setting.Denom, true); err != nil {
+			panic(err)
+		}
+	}
+
+	for _, collateral := range genState.Collateral {
+		borrower, err := sdk.AccAddressFromBech32(collateral.Address)
+		if err != nil {
+			panic(err)
+		}
+
+		if err = k.SetCollateralAmount(ctx, borrower, collateral.Amount); err != nil {
+			panic(err)
+		}
+	}
+
+	for _, reserve := range genState.Reserves {
+		if err := k.SetReserveAmount(ctx, reserve); err != nil {
+			panic(err)
+		}
+	}
+
+	if err := k.SetLastInterestTime(ctx, sdk.NewInt(genState.LastInterestTime)); err != nil {
```

Comment:
Should we an int64. Why are we using `sdk.Int` here?

Reply:
I'm storing it as an sdk.Int in state so I can use `Marshal` and `Unmarshal` methods - is it ok to keep that?

If so, I can still alter the signatures of `k. SetLastInterestTime` and `k.GetLastInterestTime` to use `int64`

Reply:
e.g.

```
		val := gogotypes.UInt64Value{}

		err := cdc.Unmarshal(bz, &val)
		if err != nil {
			panic(err)
		}
		```